src/components/textBlocks/ListItem.vue

```diff
@@ -0,0 +1,395 @@
+import { mapStores } from 'pinia'
+import mitt from 'mitt'
+import useMainStore from '../../store/mainStore.js'
+import { getShares, shareTextBlock, unshareTextBlock } from '../../service/TextBlockService.js'
+import TextEditor from '../TextEditor.vue'
+import { showError, showSuccess } from '@nextcloud/dialogs'
+import AccountMultiple from 'vue-material-design-icons/AccountMultiple.vue'
+import IconClose from 'vue-material-design-icons/Close.vue'
+import IconCheck from 'vue-material-design-icons/Check.vue'
+import IconPencil from 'vue-material-design-icons/Pencil.vue'
+import debounce from 'lodash/fp/debounce.js'
+import { ShareType } from '@nextcloud/sharing'
+import { generateOcsUrl } from '@nextcloud/router'
+import { getCurrentUser } from '@nextcloud/auth'
+
+import axios from '@nextcloud/axios'
+
+export default {
+	name: 'ListItem',
+	components: {
+		NcActionButton,
+		NcLoadingIcon,
+		NcSelect,
+		NcDialog,
+		TextEditor,
+		NcButton,
+		NcAvatar,
+		NcInputField,
+		AccountMultiple,
+		IconClose,
+		IconPencil,
+		IconCheck,
+		NcListItem,
+	},
+	props: {
+		textBlock: {
+			type: Object,
+			required: true,
+		},
+		shared: {
+			type: Boolean,
+			default: false,
+		},
+		isViewMode: {
+			type: Boolean,
+			default: false,
+		},
+		picked: {
+			type: Boolean,
+			default: false,
+		},
+	},
+	data() {
+		return {
+			shares: [],
+			localTextBlock: Object.assign({}, this.textBlock),
+			editModalOpen: false,
+			loading: false,
+			saveLoading: false,
+			share: null,
+			suggestions: [],
+			bus: mitt(),
+		}
+	},
+	computed: {
+		...mapStores(useMainStore),
+		options() {
+			return this.suggestions.filter(suggestion => !this.shares.find(share => share.name === suggestion.shareWith) && suggestion.shareWith !== getCurrentUser().uid)
+		},
+		sortedShares() {
+			return [...this.shares].sort((a, b) => {
+				if (a.type === 'user' && b.type === 'group') {
+					return -1
+				}
+				if (a.type === 'group' && b.type === 'user') {
+					return 1
+				}
+				return 0
+			})
+		},
+	},
+	async mounted() {
+		if (!this.shared && !this.isViewMode) {
+			this.shares = await getShares(this.textBlock.id)
+		}
+	},
+	methods: {
+		async deleteTextBlock() {
+			await this.mainStore.deleteTextBlock({ id: this.textBlock.id }).then(() => {
+				showSuccess(t('mail', 'TextBlock deleted'))
+			}).catch(() => {
+				showError(t('mail', 'Failed to delete textBlock'))
+			})
+		},
+		async shareTextBlock(sharee) {
+			await shareTextBlock(this.textBlock.id, sharee.shareWith, sharee.shareType === ShareType.User ? 'user' : 'group').then(() => {
+				this.shares.push({ shareWith: sharee.shareWith, type: sharee.isNoUser ? 'group' : 'user', displayName: sharee.displayName })
+				showSuccess(t('mail', 'TextBlock shared with {sharee}', { sharee: sharee.shareWith }))
+				this.share = null
+			}).catch(() => {
+				showError(t('mail', 'Failed to share textBlock with {sharee}', { sharee: sharee.shareWith }))
+			})
```

Comment:
mix of async syntax and .then. Refactor to just async and use a try-catch for error handling.